Sources/Fuzzilli/Core/MutationEngine.swift

```diff
@@ -24,58 +24,41 @@ public enum CrashBehaviour: String {
 public class MutationEngine: ComponentBase, FuzzEngine {
     /// Common prefix of every generated program. This provides each program with several variables of the basic types
     private var prefix: Program
-    
+
     private var shouldPreprocessSamples: Bool {
         // Programs are only preprocessed (have a prefix added to them etc.) if there is no minimization
         // limit (i.e. programs in the corpus are minimized as much as possible). Otherwise, we assume that
         // enough script content is available due to the limited minimization.
         return fuzzer.config.minimizationLimit == 0
     }
 
-    /// Prefix "template" to use. Every program taken from the corpus
-    /// is prefixed with some code generated from this before mutation.
-    private let programPrefixGenerators: [CodeGenerator] = [
-        CodeGenerators.get("IntegerGenerator"),
-        CodeGenerators.get("StringGenerator"),
-        CodeGenerators.get("BuiltinGenerator"),
-        CodeGenerators.get("FloatArrayGenerator"),
-        CodeGenerators.get("IntArrayGenerator"),
-        CodeGenerators.get("ArrayGenerator"),
-        CodeGenerators.get("ObjectGenerator"),
-        CodeGenerators.get("ObjectGenerator"),
-    ]
-    
     // The number of consecutive mutations to apply to a sample.
     private let numConsecutiveMutations: Int
-    
-    // Mutators to use.
-    private let mutators: WeightedList<Mutator>
 
-    public init(mutators: WeightedList<Mutator>, numConsecutiveMutations: Int) {
+    public init(numConsecutiveMutations: Int) {
         self.prefix = Program()
-        self.mutators = mutators
         self.numConsecutiveMutations = numConsecutiveMutations
-        super.init(name: "MutationFuzzer")
+        super.init(name: "MutationEngine")
     }
-    
+
     override func initialize() {
-        prefix = makePrefix()
-        
+        prefix = generateProgramPrefix()
+
         // Regenerate the common prefix from time to time
         if shouldPreprocessSamples {
             fuzzer.timers.scheduleTask(every: 15 * Minutes) {
-                self.prefix = self.makePrefix()
+                self.prefix = self.generateProgramPrefix()
             }
         }
-        
+
         if fuzzer.config.logLevel.isAtLeast(.info) {
             fuzzer.timers.scheduleTask(every: 15*Minutes) {
-                let stats = self.mutators.map({ "\($0.name): \(String(format: "%.2f%%", $0.correctnessRate * 100))" }).joined(separator: ", ")
+                let stats = self.fuzzer.mutators.map({ "\($0.name): \(String(format: "%.2f%%", $0.stats.correctnessRate * 100))" }).joined(separator: ", ")
```

Comment:
I think this entire block (lines 54 - 59) should probably be moved into `Fuzzer` since that's where the Mutators are now. That way, with the MultiEngine enabled, these messages won't be printed twice (by the HybridEngine and this one)